File: javascript-closure-angular/API/Client/ResponseTimeFilterProperties.js
goog.provide('API.Client.ResponseTimeFilterProperties');

/**
 * @record
 */
API.Client.ResponseTimeFilterProperties = function() {}

/**
 * @type {!number}
 * @export
 */
API.Client.ResponseTimeFilterProperties.prototype.travelTime;

/**
 * @type {!number}
 * @export
 */
API.Client.ResponseTimeFilterProperties.prototype.distance;

/**
 * @type {!Array<!API.Client.ResponseDistanceBreakdownItem>}
 * @export
 */
API.Client.ResponseTimeFilterProperties.prototype.distanceBreakdown;

/**
 * @type {!API.Client.ResponseFares}
 * @export
 */
API.Client.ResponseTimeFilterProperties.prototype.fares;

/**
 * @type {!API.Client.ResponseRoute}
 * @export
 */
API.Client.ResponseTimeFilterProperties.prototype.route;

File: scalaz/src/main/scala/org/openapitools/client/api/ResponseTimeFilterProperties.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ResponseTimeFilterProperties._

case class ResponseTimeFilterProperties (
  travelTime: Option[Integer],
distance: Option[Integer],
distanceBreakdown: Option[List[ResponseDistanceBreakdownItem]],
fares: Option[ResponseFares],
route: Option[ResponseRoute])

object ResponseTimeFilterProperties {
  import DateTimeCodecs._

  implicit val ResponseTimeFilterPropertiesCodecJson: CodecJson[ResponseTimeFilterProperties] = CodecJson.derive[ResponseTimeFilterProperties]
  implicit val ResponseTimeFilterPropertiesDecoder: EntityDecoder[ResponseTimeFilterProperties] = jsonOf[ResponseTimeFilterProperties]
  implicit val ResponseTimeFilterPropertiesEncoder: EntityEncoder[ResponseTimeFilterProperties] = jsonEncoderOf[ResponseTimeFilterProperties]
}
